tests/nested/manual/passphrase-support-on-hybrid/task.yaml

```diff
@@ -0,0 +1,196 @@
+  remote.exec "sudo test -e /var/lib/snapd/device/fde/marker"
+  remote.exec "sudo test -e /var/lib/snapd/device/fde/marker"
+  remote.exec "sudo blkid /dev/disk/by-label/ubuntu-data-enc" | MATCH crypto_LUKS
+
+  # Check disk mappings.
+  # TODO: no ubuntu-save right now because:
+  #       "ERROR cannot store device key pair: internal error: cannot access device keypair manager if ubuntu-save is unavailable"
+  #DISK_MAPPINGS=(/run/mnt/ubuntu-save/device/disk-mapping.json
+  #               /run/mnt/data/var/lib/snapd/device/disk-mapping.json)
+  DISK_MAPPINGS=(/run/mnt/data/var/lib/snapd/device/disk-mapping.json)
+  for DM in "${DISK_MAPPINGS[@]}"; do
+      remote.exec "sudo cat $DM" > mapping.json
+      gojq -r '.pc."structure-encryption"."ubuntu-save".method' < mapping.json | MATCH LUKS 
+      gojq -r '.pc."structure-encryption"."ubuntu-data".method' < mapping.json | MATCH LUKS
+  done
+
+  # Check that on an already provisioned system the API will give a
+  # sensible reason why the system cannot be installed without further
+  # action.
+  remote.exec "sudo snap debug api /v2/systems/classic" > system
+  gojq '.result."storage-encryption".support' < system | MATCH "unavailable"
+  gojq '.result."storage-encryption"."unavailable-reason"' < system | MATCH "not encrypting device storage as checking TPM gave: the TPM is in DA lockout mode"
+
+  # refresh rebooting snap
+  # $1: path to snap file
+  # $2: snap name
+  # $3: reboot action ("reboot"/"no-reboot")
+  refresh_rebooting_snap()
+  {
+      local snap_filename=$1
+      local snap_name=$2
+      local reboot_action=$3
+
+      boot_id=$(tests.nested boot-id)
+
+      printf "Test installing snap from file %s\n" "$snap_filename"
+      remote.push "$snap_filename"
+      # install will exit when waiting for the reboot
+      remote.exec sudo snap install --dangerous "$snap_filename" | MATCH "Task set to wait until a system restart allows to continue"
+
+      # Check that a reboot notification was setup.
+      remote.exec test -f /run/reboot-required
+      remote.exec cat /run/reboot-required.pkgs | MATCH "snap:${snap_name}"
+      # Check that no reboot has been scheduled, then force a reboot
+      remote.exec not test -f /run/systemd/shutdown/scheduled
+
+      if [ "$reboot_action" = "reboot" ]; then
+          local log_file="$NESTED_LOGS_DIR"/serial.log
+          # Clear old log file to avoid matching passphrase prompt from previous boot.
+          echo "" > "$log_file"
+          remote.exec sudo reboot || true
+          remote.wait-for reboot --wait 1 -n 200 --expect-passphrase "ubuntu" --log-file "$log_file" "$boot_id"
+          remote.exec sudo snap watch --last=install
+      fi
+  }
+  # Ensure update-notifier-common is installed so that reboot notification works.
+  remote.exec "sudo apt install -y update-notifier-common"
+
+  # Initial reseal count.
+  remote.exec sudo cat /var/lib/snapd/device/fde/boot-chains > boot-chains-before.json
```

Comment:
I think we will need to remove this file at some point. I would prefer if we stop adding tests that rely on it.

This file is mostly used to cache and avoid recomputing input for the pcr profiles. But since we have a FDE state now, we will be able to looking into ditching it. So please, do not use that.

You could query the state and look if the  pcr profile changed maybe instead.